### PR TITLE
Fix renderpass memory leak

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -10054,43 +10003,15 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateRenderPass(VkDevice device, const VkRenderP
...
-    }
-    my_data->renderPassMap.clear();
-}
-
 static bool VerifyFramebufferAndRenderPassLayouts(layer_data *dev_data, GLOBAL_CB_NODE *pCB, const VkRenderPassBeginInfo *pRenderPassBegin) {
     bool skip_call = false;
-    const VkRenderPassCreateInfo *pRenderPassInfo = dev_data->renderPassMap[pRenderPassBegin->renderPass]->pCreateInfo;
-    const safe_VkFramebufferCreateInfo framebufferInfo = dev_data->frameBufferMap[pRenderPassBegin->framebuffer]->createInfo;
+    auto const pRenderPassInfo = getRenderPass(dev_data, pRenderPassBegin->renderPass)->createInfo.ptr();
+    auto const & framebufferInfo = dev_data->frameBufferMap[pRenderPassBegin->framebuffer]->createInfo;
```
Comment (tobine) [inline layers/core_validation.cpp]: Not important for this PR, but we should migrate this to use getFramebuffer() helper similar to line above it.

File: layers/core_validation_types.h [hunk 2]
```diff
@@ -297,14 +297,14 @@ struct DAGNode {
 
 struct RENDER_PASS_NODE : public BASE_NODE {
     VkRenderPass renderPass;
-    VkRenderPassCreateInfo const *pCreateInfo;
+    safe_VkRenderPassCreateInfo createInfo;
```
Comment (tobine) [inline layers/core_validation_types.h]: You've kept existing camelCase in variable declarations throughout the change. Please try to migrate to lowercase w/ "_" going fwd per coding standard. I know it's the least of our concerns but making such updates during cleanup like this is a good way to slowing get the code consistent.

**Comment by chrisforbes** [comment]
Clean on the farm.

**Comment by chrisforbes** [comment]
Pushed to master.

**Pull request closed without merging**